### ISSUE TITLE
ColiseuPlayer: Singleton

### ISSUE THREAD

**Issue by ricardopereira** [ricardopereira/ColiseuPlayer#3]

**Comment by SentulAsia** [comment]
@ricardopereira does this involve dispatch_async?

**Comment by ricardopereira** [comment]
Yes for different queues but it's a premature optimisation :grin: 
Not necessary for now.

**Comment by SentulAsia** [comment]
@ricardopereira for singleton, should I add `static shared` instance and make init as private?

**Comment by ricardopereira** [comment]
@SentulAsia Yes, `static shared` and the `init` as `@unavailable`.

**Comment by ricardopereira** [comment]
Well, now that I'm thinking about it, maybe we don't need a singleton and we can assume the latest `ColiseuPlayer` instance will always take control of the audio.

@SentulAsia  What your thoughts about this?

**Comment by SentulAsia** [comment]
As discussed in #26 :

@ricardopereira static shared is another way to solve this problem...
Multiple audio instance fighting to play audio on the same speaker would be troublesome...
My previous app have a playlist audio which is consistently playing, then when some button is tapped, there is an effect sound (a short 1 sec sound)...
It turns out we need to have only 1 instance of audio object or else, the latest object will get to play audio, while the older instance object will stay in silence...
